### PR TITLE
fix: account resources persistence issues

### PR DESCRIPTION
**Pull request by UnseenWizzard**
#### What this PR does / Why we need it:
Fixes the following issues: 

- **loader**: creating wrong defined environment policy passes without error
- **writer**:
  - for users without any group, group segment is present
  - write process panics for groups without policy

#### Special notes for your reviewer:
/

#### Does this PR introduce a user-facing change?
no, not released and just bug fixes

### CODE REVIEW

File: pkg/persistence/account/internal/types/types.go [hunk 1]
```diff
@@ -16,68 +16,101 @@
...
+
+const (
+	ReferenceType          = "reference"
+	PolicyLevelAccount     = "account"
+	PolicyLevelEnvironment = "environment"
+)
+
 type (
-	PolicyLevel = any // either PolicyLevelAccount or PolicyLevelEnvironment is allowed
-	PolicyRef   = any // either string or Reference is allowed
```
Comment (UnseenWizzard) [inline pkg/persistence/account/internal/types/types.go:21]: note: I think we should reflect the change to explicit Level and Reference type(s) for the in-memory structs as well as a followup.

**Review by jskelin — APPROVED** [review]

**Review by jskelin — APPROVED** [review]

File: pkg/persistence/account/internal/types/types.go (hunk 1 above)
Comment (warber) [reply pkg/persistence/account/internal/types/types.go:21]: in memory types already have a reference type, if you mean that

**Comment by warber** [comment]
lgtm, Nit: i think we can get rid of the two sentinel errors in errors.go as it is nowhere else used and/or needed. wdyt?

File: pkg/persistence/account/internal/types/types.go (hunk 1 above)
Comment (UnseenWizzard) [reply pkg/persistence/account/internal/types/types.go:21]: ah nice. Didn't look at them for a while

**Comment by UnseenWizzard** [comment]
@warber the sentinel errors are used in a few test asserts.. generally agree we probably don't need them, but IMO they don't hurt to have?

**Pull request merged**